presto-main/src/main/java/com/facebook/presto/sql/planner/QueryPlanner.java

```diff
@@ -500,23 +500,21 @@ private PlanBuilder window(PlanBuilder subPlan, QuerySpecification node)
             Window window = windowFunction.getWindow().get();
 
             // Extract frame
-            WindowFrame.Type frameType = WindowFrame.Type.RANGE;
-            FrameBound.Type frameStartType = FrameBound.Type.UNBOUNDED_PRECEDING;
-            FrameBound.Type frameEndType = FrameBound.Type.CURRENT_ROW;
+            WindowFrame.Type frameType = null;
+            FrameBound.Type frameStartType = null;
+            FrameBound.Type frameEndType = null;
             Expression frameStart = null;
             Expression frameEnd = null;
 
-            if (window.getFrame().isPresent()) {
-                WindowFrame frame = window.getFrame().get();
-                frameType = frame.getType();
+            WindowFrame frame = window.getFrame();
```

Comment:
Move it before https://github.com/Teradata/presto/pull/304/commits/f5a57febdbef7b62f1ef628aaede4c40d666c32b#diff-21d31cb9d487772413eff4f67b8555deR503 to avoid unnecessary null-initialization of e.g. `frameType`.